source/api_c/include/c_api.h

```diff
@@ -317,6 +365,51 @@ extern void DP_DeepPotCompute2(DP_DeepPot* dp,
                                double* atomic_energy,
                                double* atomic_virial);
 
+/**
+ * @brief Evaluate the energy, force, magnetic force and virial by using a DP
+ *with spin input. (double version)
+ * @version 2
+ * @param[in] dp The DP to use.
+ * @param[in] nframes The number of frames.
+ * @param[in] natoms The number of atoms.
+ * @param[in] coord The coordinates of atoms. The array should be of size natoms
+ *x 3.
+ * @param[in] spin The spins of atoms, [0, 0, 0] if no spin. The array should be
+ *of size nframes x natoms x 3.
+ * @param[in] atype The atom types. The array should contain natoms ints.
+ * @param[in] box The cell of the region. The array should be of size 9. Pass
+ *NULL if pbc is not used.
+ * @param[in] fparam The frame parameters. The array can be of size nframes x
+ *dim_fparam.
+ * @param[in] aparam The atom parameters. The array can be of size nframes x
+ *dim_aparam.
+ * @param[out] energy Output energy.
+ * @param[out] force Output force. The array should be of size natoms x 3.
+ * @param[out] force_mag The magnetic force on each atom.
```

Comment:
Since the array needs to allocate the memory in advance, it's important to document the size.